### PR TITLE
chore: move GLTFs texture resize to the GPU for desktop client

### PR DESCRIPTION
**Pull request by pravusjif**
* In many parts of the world we are resizing every texture loaded that is bigger than 1024x1024. We even have lots of textures in the first base wearables that get resized every time we enter the world if any avatar is using them. (With Asset Bundles this resizing is avoided as those textures were already resized when converting the asset to AB).

* If we can move to the GPU the textures resize we do in runtime for the Desktop Client, it will run smoother now that we have removed the runtime compressions for the Desktop Client as well.

* This optimization of using `Graphics.CopyTexture()` (copying everything directly in the GPU without passing through the CPU) instead of reading from the `RenderTexture` with `Texture2d.ReadPixels()` + `Texture2d.Apply()` that's quite expensive, can only be done for the Desktop client as it supports using the CopyTexture() method and the Web client doesn't.

A place where textures resizing can be seen is at:
https://play.decentraland.zone/?renderer-branch=chore/MoveGLTFsTextureResizeToGPUForDesktop&DISABLE_ASSET_BUNDLES&DISABLE_WEARABLE_ASSET_BUNDLES&position=-32,68&LOS=2

If these images can be seen, the resizing worked fine:
chrome_zBmTglA1LL

### CODE REVIEW

**Review by BrianAmadori — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/TextureHelpers/TextureHelpers.cs [hunk 1]
```diff
@@ -54,7 +66,10 @@ public static Texture2D Resize(Texture2D source, int newWidth, int newHeight)
     public static Texture2D CopyTexture(Texture2D sourceTexture)
     {
         Texture2D texture = new Texture2D(sourceTexture.width, sourceTexture.height, sourceTexture.format, false);
-        Graphics.CopyTexture(sourceTexture, texture); // TODO: does this work in WebGL?
+        
+        // TODO: WebGL doesn't support this, should we add the ReadPixels/GetPixels approach as in Resize() ?
+        Graphics.CopyTexture(sourceTexture, texture);
```
Comment (kuruk-mm) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/TextureHelpers/TextureHelpers.cs:71]: Should we change this with:

```
        bool supportsGPUTextureCopy = SystemInfo.copyTextureSupport != CopyTextureSupport.None;
        if (supportsGPUTextureCopy)
        {
            Graphics.CopyTexture(rt, nTex);
        }
        else
        {
            nTex.ReadPixels(new Rect(0, 0, newWidth, newHeight), 0, 0);
            nTex.Apply();
        }
```

and in the resize function just use:

```
    Graphics.CopyTexture(rt, nTex);
```

?

**Review by kuruk-mm — APPROVED** [review]
Great! I let you a little comment

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/TextureHelpers/TextureHelpers.cs (hunk 1 above)
Comment (pravusjif) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/TextureHelpers/TextureHelpers.cs:71]: Yeah that's why I updated the comment on that line, I just talked to Brian about this, I'll apply that change today before merging.

Comment (pravusjif) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/TextureHelpers/TextureHelpers.cs]: done!

**Pull request merged**